### ISSUE TITLE
Add documentation about what array views are and where they come up

### ISSUE THREAD

**Issue by stonea** [chapel-lang/chapel#19177]
The reindex returns an array view over the domain.  Looking at our module code I can see three different kinds of array views:

 `ArrayViewRankChange.chpl  ArrayViewReindex.chpl     ArrayViewSlice.chpl`

So it sounds like we we also introduce "array views" in other contexts.

However near as I can tell, nowhere in the doc do we explain what an array view is.

On the one hand, it to me like its obvious enough to explain that this is some object that acts as a reindexing of an existing array while referencing that original array (so no data gets copied), on the other if we're going to introduce the term "array view" in the doc we might want to have some formalish definition of what that is.

**Comment by bradcray** [comment]
For reference, given an array `A: [1..n, 1..n] real;`  the three current types of array views are:

* slices:  `A[2..<n, 2..<n]`
* rank change slices:  `A:[2, 1..n]`
* reindexing:  `A.reindex[0..<n, 0..<n]`

The thing that makes these array views is that they are meant to act like normal arrays yet to alias an existing array's memory.  E.g., capturing any of the expressions in a reference:

```chapel
ref Aalias = [array view expression];
```

and writing to that `Aalias` variable will change the original array's elements' values.  Of these three forms, slices are used most prominently, rank-change next, and reindexing last.

I think we didn't document these because we viewed the "array view" as more of an implementation detail than something that a user would need to know about (similar to how we don't document the `DefaultRectangularArr` class for users, for instance).  But I'm not sure how successful we've been at that or not, nor whether it was the right decision.

**Comment by stonea** [comment]
> I think we didn't document these because we viewed the "array view" as more of an implementation detail than something that a user would need to know about (similar to how we don't document the DefaultRectangularArr class for users, for instance). But I'm not sure how successful we've been at that or not, nor whether it was the right decision.

Yeah I'm not totally convinced it's worth explicitly documenting either.

The reason I brought it had to do with how we document `reindex` (in https://chapel-lang.org/docs/main/language/spec/arrays.html):

> "Return an array view over a new domain"

And it bothers me that we use the term "array view" but I can't find a definition of that anywhere. 

Another (simpler) solution would just be to reword it \ spell out what array view means in the definition for reindex. This could be as simple as adding a parenthetical sentence or two.

So change:

"Return an array view over a new domain."

to something like:

"Returns an array view (an object with the same interface as an array, which references the exact same data as the original array and can be used to update that data, but reindexes it into a new domain)."
